### PR TITLE
Added ⚡️ content to `Private Key Management` > `Backups`

### PR DESCRIPTION
**Pull request by sbddesign**
☁️ [Cloud Backup - Deploy Preview](https://deploy-preview-559--sad-borg-390916.netlify.app/guide/private-key-management/cloud-backup/)
📝 [Manual Backup - Deploy Preview](https://deploy-preview-559--sad-borg-390916.netlify.app/guide/private-key-management/manual-backup/)

Here are some updates I made to the Cloud Backups and Manual Backups in the Private Key Management chapter. Fixes for #544 and #555.

Most of these are grammatical and typographical changes, but I mentioned Lightning in the cloud backup chapter.

Since this chapter is focusing specifically on the private key, I don't feel like there is a lot of differentiation between on-chain and off-chain; in either scenario, it's critical the private key is stored and backed up safely.

That being said, we should absolutely update this content once Lightning Signer is farther along in development. I'm excited about external signing APIs for LN, but I don't feel comfortable writing about it at the moment while it's still so early in the development process. (Or perhaps that content should go in the External Signing Device chapter).

### CODE REVIEW

File: guide/private-key-management/cloud-backup.md
Comment (pavlenex) [inline guide/private-key-management/cloud-backup.md:44]: "it would not take much work" - is there any chance we can explain this a bit? What kind of work process is involved in this?

File: guide/private-key-management/cloud-backup.md
Comment (pavlenex) [inline guide/private-key-management/cloud-backup.md:93]:
```suggestion
You can provide an extra layer of security for the user's private key by encrypting it before sending it to the cloud. One option would be for the user to choose an encryption passphrase. Another would be to have a third-party hold the encryption key so that neither the cloud provider nor the encryption key holder would have access to all the information to use the private key. See [this article](https://blog.keys.casa/casa-keymaster-security-mobile-key-overview/) for an example.
```

File: guide/private-key-management/manual-backup.md
Comment (pavlenex) [inline guide/private-key-management/manual-backup.md:38]:
```suggestion
Manual backup of the [recovery phrase]({{ '/guide/glossary/#recovery-phrase' | relative_url }}) has been the most common private key management scheme by far since its proposal with [BIP39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) in 2013. If you have used any [non-custodial]({{ '/guide/glossary/wallet/#non-custodial--custodial-wallet' | relative_url }}) Bitcoin application, you are likely to have experienced the onboarding requirements of manual backups.
```

**Review by pavlenex — COMMENTED** [review]
LGTM, just a few nits I found.

File: guide/private-key-management/cloud-backup.md
Comment (sbddesign) [reply guide/private-key-management/cloud-backup.md:44]: That's an assumption I am making (perhaps wrong). What I'm trying to express is that if the developer has already built a way for the wallet app to communicate with a cloud provider for frequent channel state updates, then it seems to me like it would not require much additional coding to also store a backup of the key as well.

Maybe something like this:

> If you have implemented cloud backups for Lightning channel state in your app, you could also use this same conection with the cloud provider to store a backup of the user's private key.

**Comment by Bosch-0** [comment]
> (Or perhaps that content should go in the External Signing Device chapter).

Have a small section on this in the External Signing Device chapter PR https://github.com/BitcoinDesign/Guide/pull/555

File: guide/private-key-management/cloud-backup.md
Comment (Bosch-0) [inline guide/private-key-management/cloud-backup.md:38]:
```suggestion
By using an cloud backup for the private key the user is NOT required to manually backup anything when setting up a new [bitcoin wallet]({{ '/guide/glossary/wallet/#bitcoin-wallet' | relative_url }}). This is sometimes also called *seedless*, as the user is never exposed to the traditional *seed*, or [recovery phrase]({{ '/guide/glossary/#recovery-phrase' | relative_url }}) that is used to initiate a wallet.
```

If we are changing this page title, like the cloud backups page in onboarding, we should be dropping the 'automatic' term for reasons outlined when adding LN content to that section.

File: guide/private-key-management/cloud-backup.md
Comment (Bosch-0) [inline guide/private-key-management/cloud-backup.md:40]: Could add some details here about a Lightning enabled daily spending wallet and how cloud backup can be handy for that.

File: guide/private-key-management/cloud-backup.md
Comment (Bosch-0) [inline guide/private-key-management/cloud-backup.md:47]:
```suggestion
First the wallet application generates the users private keys. Then, the keys are encrypted and backed up to a location convenient and safe for the user, yet hard for a malicious third-party to gain access to.
```

Wondering if we should put these steps in a list. As well as add a third step around if the user needs to use the backup they do xyz.

File: guide/private-key-management/cloud-backup.md
Comment (Bosch-0) [inline guide/private-key-management/cloud-backup.md:88]: I bumped the size of the subheadings up 1 in the `Best practice` section in my external signers PR, looks a little more legible imo.

File: guide/private-key-management/manual-backup.md
Comment (Bosch-0) [inline guide/private-key-management/manual-backup.md:38]:
```suggestion
Manual backup of the [recovery phrase]({{ '/guide/glossary/#recovery-phrase' | relative_url }}) has been the most common private key management scheme since its proposal with [BIP39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) in 2013. If you have used any [non-custodial]({{ '/guide/glossary/wallet/#non-custodial--custodial-wallet' | relative_url }}) Bitcoin application, you are likely to have experienced the onboarding requirements of manual backups.
```

File: guide/private-key-management/manual-backup.md
Comment (Bosch-0) [inline guide/private-key-management/manual-backup.md:40]:
```suggestion
When creating a new wallet, users will be asked to manually write down a backup of a 12 or 24 word recovery phrase to a *safe place*. Often, as the next step, it will ask you to verify that you did save it by having you input the recovery phrase in the correct order. Additionally, some wallets may use a [passphrase]({{'/guide/glossary/#passphrase' | relative_url}}) that can be defined by the user.
```

**Review by Bosch-0 — COMMENTED** [review]
Looking good, left some comments :)

**Comment by danielnordh** [comment]
What's the rational for dropping 'Automatic' from the scheme name / page title?

IMO, automatic is a defining feature of this scheme, requiring NO user interaction to achieve the backup.
It's also described as such on the other pages, and in the graphics on the Overview page.

I would like to retain the name 'Automatic cloud backup'.

File: guide/private-key-management/cloud-backup.md
Comment (danielnordh) [reply guide/private-key-management/cloud-backup.md:38]: I'm against dropping 'Automatic' as it's a defining feature of this scheme.

File: guide/private-key-management/cloud-backup.md
Comment (danielnordh) [reply guide/private-key-management/cloud-backup.md:40]: Agree, an automatic cloud backup scheme is likely the best fit for many Lightning wallets as it enables continuous backup of channel states.

File: guide/private-key-management/cloud-backup.md
Comment (danielnordh) [reply guide/private-key-management/cloud-backup.md:38]: I'm against dropping 'Automatic', it's a defining feature of this scheme.

File: guide/private-key-management/cloud-backup.md
Comment (danielnordh) [reply guide/private-key-management/cloud-backup.md:40]: Agree, automatic cloud backups are likely the best fit for many Lightning wallets as it enables continuous backups of the channel states.

**Review by danielnordh — CHANGES_REQUESTED** [review]
Retain 'Automatic cloud backup' as the name / title.

**Comment by pavlenex** [comment]
> What's the rational for dropping 'Automatic' from the scheme name / page title?
> 
> IMO, automatic is a defining feature of this scheme, requiring NO user interaction to achieve the backup. It's also described as such on the other pages, and in the graphics on the Overview page.
> 
> I would like to retain the name 'Automatic cloud backup'.

I am not sure what rationale was for this, as I kinda liked the sound of it as well. Just for transparency, the `automatic` has been dropped from the `Onboarding` already. So if we were to retain it, we'd have to do it for that section as well.

**Comment by danielnordh** [comment]
> I am not sure what rationale was for this, as I kinda liked the sound of it as well. Just for transparency, the `automatic` has been dropped from the `Onboarding` already. So if we were to retain it, we'd have to do it for that section as well.

If we say 'manual' for one type, we should say 'automatic' for the other. 
It seems to be inconsistent in the Onboarding section, both with and without Automatic:
image

**Comment by sbddesign** [comment]
@danielnordh and @pavlenex we removed "automatic" in #541 for the Onboarding section. Discussion started in the initial Google Drive doc for that draft. It was @Bosch-0 suggestion, basically that there are many situations where a user may consent to having it backed up in the cloud (if the app prompts them first about it, or if the app must check for permissions first). Any sections where it still says "automatic" in Onboarding are my mistake and just things I missed.

But I also see the other side to this; it's automatic because it doesn't involve writing down a bunch of words.

I could go both ways. Automatic or not automatic, I think this is just kind of an issue of semantics and I don't think it affects the meat of the content. I'll be happy with it either way. Whatever we choose, I'll make it consistent in Onboarding as well as Private Key Management.

File: guide/private-key-management/cloud-backup.md
Comment (sbddesign) [reply guide/private-key-management/cloud-backup.md:47]: I'm not sold on this. As far as I understand, all of the "private keys" do not need to be backed up because they can all be deterministically generated from the parent private key. Though I do think we have a slight confusing discrepancy when we say "keypair" vs. "key" so I think I'll just make that "private key".

I don't think the list is necessary, it's a simple tiny paragraph and we explain how the users restores a few lines down.

File: guide/private-key-management/cloud-backup.md
Comment (sbddesign) [reply guide/private-key-management/cloud-backup.md:88]: Done

File: guide/private-key-management/cloud-backup.md
Comment (sbddesign) [reply guide/private-key-management/cloud-backup.md]: Done

File: guide/private-key-management/manual-backup.md
Comment (sbddesign) [reply guide/private-key-management/manual-backup.md:38]: I think manual backups have been the most common by far. There is a difference between being "most common" (say, 51%) and being "by far most common" (say, 75-90%). I have no data to backup the claim, just going off what I have experienced. Thoughts anybody?

File: guide/private-key-management/cloud-backup.md
Comment (sbddesign) [reply guide/private-key-management/cloud-backup.md]: Done. I rearranged the order of the paragraph to flow a little better with the addition of the LN content.

**Comment by danielnordh** [comment]
Apologies for missing the original discussion in the word doc.

> I could go both ways. Automatic or not automatic, I think this is just kind of an issue of semantics and I don't think it affects the meat of the content. I'll be happy with it either way. Whatever we choose, I'll make it consistent in Onboarding as well as Private Key Management.

Although I can see a way of looking at this as semantics, I much prefer to be really clear here that this scheme is supposed to be foolproof, and distinct from a 'manual cloud backup' as shown on the [Bitcoin backups page](https://bitcoin.design/guide/private-key-management/backups/#encrypted-cloud-backup)

**Comment by pavlenex** [comment]
So the blocker for this PR is adding `automatic` into the cloud backup. Is everyone okay with this?
If we are, shouldn't be a big deal to add `automatic` back, and in the next PR we can address and and add it back for consistency in the onboarding page.

What do you guys think? @sbddesign @Bosch-0 @danielnordh

**Comment by GBKS** [comment]
I also prefer "Automatic cloud backup". Just easier to get the primary benefit.

To throw another ring in the game, how about calling it "Automatic backup"? 😉

**Comment by danielnordh** [comment]
I'm OK with 'Automatic backup', as it is a clear counterpoint to 'Manual backup'.
I still think 'Automatic cloud backup' contains more signal 🤷‍♂️

**Comment by Bosch-0** [comment]
My main reasoning behind removing automatic is that nothing in Bitcoin should be automatic entirely - it goes against the ethos and abstracts away sovereignty for UX imo. Recovery phrase cloud backups should always be manually initiated, even if that just means flicking a toggle. Pushing designers towards making a crucial step like recovery phrase backup being completely automatic could be troublesome. I don't think we are doing that but it could be interpreted in that way by some. 

To throw another spanner in the works I'd keep 'Cloud backup' and rename manual backup to 'Physical backup.' I like naming convention where it's descriptive of where it is backed up rather than how. Recovery phrases should be backed up physically, manual backup could mean manually backing up physically or digitally.

**Comment by GBKS** [comment]
@Bosch-0 I don't think we're pushing too hard on full automation. Users still need to give consent before the app "automatically" backs up the recovery phrase, they get properly informed beforehand, are asked to give their consent, and are offered the manual option if they don't want to do it. If we're concerned people go too far with automation, we can be more explicit about this education/consent.

Regarding the term "Physical", what about putting your recovery phrase in a service like 1Password? Is that secure enough to recommend?

As you may know by now from my comments on various other conversations, I just wouldn't get hung up on single words. They can only capture so much meaning, and interpreted differently based on context. Let's make sure the totality of this writing and imagery captures what we want to say. YMMV YOLO.

**Comment by Bosch-0** [comment]
Alright ill concede on that point, I agree it's not a major issue. Cloud backup also just gets to the point, automatic doesn't really need to be there. I just like more concise terms personally. 

> Regarding the term "Physical", what about putting your recovery phrase in a service like 1Password? Is that secure enough to recommend?

I would not recommend this, I think recovery phrases should always be physical. Keyloggers, data on clipboard, malware, breaches to password manager servers, users not using good master passwords for the manager - lots of edge cases that are all avoided writing it down physically.

**Comment by pavlenex** [comment]
Let's try to wrap this one up just by adding back the `automatic cloud backup` term. I'll create a new PR and re-visit what we did in #541 and make sure it's consistent.

@sbddesign can you do that before the call today so we merge this one in? It seems that's the only blocker and the PR has had quite a bit of reviews already. I think we can discuss some structural changes you and @Bosch-0 talked about in a bit later stage, once we do #551 where lots of these will have to be re-visited, depending on how we may want to structure the content.

**Comment by sbddesign** [comment]
Yeah, I can get it updated before the call today.

**Comment by pavlenex** [comment]
@sbddesign Thanks for quickly updating those. Can you update the URL's as well so we can try to get this one merged before the call and have virtual beers to celebrate the section completing.

**Comment by sbddesign** [comment]
> @sbddesign Thanks for quickly updating those. Can you update the URL's as well so we can try to get this one merged before the call and have virtual beers to celebrate the section completing.

Are you talking about this URI?

`https://bitcoin.design/guide/private-key-management/cloud-backup/`

I *can* change it....but it looks to me like the URI is already called `/cloud-backups` on the existing Guide. It's been that way for a while; that was not me. So it will probably be less disruptive if we leave it as is.

File: guide/private-key-management/cloud-backup.md
Comment (johnsBeharry) [reply guide/private-key-management/cloud-backup.md:38]: I like the idea of a more generic on-device, and cloud backup that can be used for various types of Bitcoin application data and style procedures (manual , automatic).

**Comment by pavlenex** [comment]
This one seems like good to go. @sbddesign LMK if you want to address any comments/suggestions before we merge this.

File: guide/private-key-management/cloud-backup.md
Comment (sbddesign) [reply guide/private-key-management/cloud-backup.md]: Done

File: guide/private-key-management/cloud-backup.md
Comment (sbddesign) [reply guide/private-key-management/cloud-backup.md]: @johnsBeharry I added something to this effect, but I could not commit this suggestion directly due to falling out of sync with more recent commits.

**Comment by sbddesign** [comment]
I implemented some tweaks from @johnsBeharry.
Ready for final review and (hopefully) merge @pavlenex.

**Review by pavlenex — DISMISSED** [review]
LGTM, after the merge I'll open up the issue to address the changes in the Onboarding. 

@danielnordh Since you've requested a change, this one requires your approval for us to merge it.

**Review by Bosch-0 — APPROVED** [review]
LGTM :)

File: guide/private-key-management/cloud-backup.md
Comment (GBKS) [inline guide/private-key-management/cloud-backup.md:95]: Encryption is already mentioned further up the page under "How it works". Was a little surprised to see it repeated at the bottom as if it hadn't been mentioned before. Encryption should probably always happen, it's just a matter of how the key is generated and managed.

File: guide/private-key-management/overview.md
Comment (GBKS) [inline guide/private-key-management/overview.md:103]:
```suggestion
   caption = "Automatic cloud backup, manual backup, signing-device and multi-key."
```

**Comment by GBKS** [comment]
ACK. Had a minor comment, and then @danielnordh needs to do another review as he has a requested change.

File: guide/private-key-management/cloud-backup.md
Comment (sbddesign) [reply guide/private-key-management/cloud-backup.md]: @GBKS That's a good point. The reason I included the paragraph is that there is another line on the page under "Cons" that reads "User needs to trust their OS and cloud storage provider".

When I read this, it sounds to me like a plaintext backup where the user is hoping that their cloud provider doesn't swipe their private key. That's why I decided to include a snippet discussing that the cloud backup can be encrypted in a way to where the cloud provider does not need to be trusted.

I just pushed a new commit where I tweaked the way this is worded and tried to group the tip box closer to the first mention of "encrypted".

**Review by Bosch-0 — APPROVED** [review]

**Review by danielnordh — APPROVED** [review]

**Pull request merged**
